app/screens/ChatbotScreen.tsx

```diff
@@ -0,0 +1,361 @@
+        .then((id) => {
+          if (id === "false") {
+            NotificationService.error(t("kyc.bot.error.wrong_sms"))
+            // wait a short amount of time to avoid screen flickering due to fast network request
+            setTimeout(() => setLoading(false), 1000)
+          } else {
+            setChatbotId(id)
+            setSMSCompleted(true)
+            requestStart(sessionId, id)
+          }
+        })
+    } else {
+      NotificationService.error(t("kyc.bot.error.missing_session_id"))
+    }
+  }
+
+  const requestStart = (id?: string, chatbotId?: string) => {
+    getStatus(id, chatbotId)
+      .then((status) => {
+        switch (status) {
+          case ChatbotStatus.INITIAL:
+            // fresh start we need to trigger first question
+            requestNextStep(chatbotStart(), undefined, chatbotId)
+            break
+          case ChatbotStatus.STARTED:
+            // already started, need to call update and parse existing pages
+            restoreSession(id, chatbotId)
+            break
+        }
+      })
+  }
+
+  const restoreSession = (id?: string, chatbotId?: string, inputPages?: ChatbotPage[], shouldDoUpdates: boolean = true): Promise<ChatbotPage[]> => {
+    if (inputPages === undefined) {
+      inputPages = pages
+    }
+    return getUpdate(id, chatbotId)
+      .then((question) => {
+        let restoredPages = chatbotRestorePages(question, settings?.language)
+        let combinedPages = inputPages?.concat(restoredPages) ?? restoredPages
+        if (shouldDoUpdates) {
+          setPages(combinedPages)
+          setAnswer(combinedPages.slice(-1)[0].answer)
+          // jump to last index
+          setPageIndex(combinedPages.length - 1)
+          setLoading(false)
+        }
+        return combinedPages
+      })
+  }
+
+  const requestEditAnswer = (newValue: string, timestamp: number, chatbotId?: string) => {
+    setRequestingNextStep(true)
+    setLoading(true)
+    requestSkip(timestamp, sessionId, chatbotId)
+      .then(() => {
+        // request update to restore all pages from KYC spider
+        return restoreSession(sessionId, chatbotId, [], false)
+          
+      })
+      .then((newPages) => {
+        // given answer is not attached to any page anymore, because of recreation of all pages
+        // therefore get recreated pages' last page answer object and fill that with correct values via requestNextStep
+        let recreatedAnswer = newPages[newPages.length-1].answer
+        if (recreatedAnswer !== undefined) {
+          requestNextStep(chatbotCreateAnswer(newValue, recreatedAnswer), recreatedAnswer, chatbotId, newPages)
+        }
+      })
+  }
+
+  const requestNextStep = (apiAnswer: ChatbotAPIAnswer, answer?: ChatbotAnswer, chatbotId?: string, inputPages?: ChatbotPage[]): Promise<ChatbotPage[]> => {
+    setRequestingNextStep(true)
+    if (inputPages === undefined) {
+      inputPages = pages
+    }
+    return nextStep(sessionId, chatbotId, apiAnswer)
```

Comment:
If we had more time, I'd design this whole logic a bit different. Seems a little convoluted. Would be more readable and bullet-proof with a more reactive code design. But it's fine for now.

Reply:
Why exactly should this function return a Promise?? Just asking to understand reasoning behind

Reply:
not needed anymore, I think there was a try to fix something by returning a Promise here, which got removed already and has been forgotten to also remove this return